pkg/kubecost/allocation.go

```diff
@@ -2131,34 +2131,179 @@ func (asr *AllocationSetRange) NewAccumulation() (*AllocationSet, error) {
 	return allocSet, nil
 }
 
-// AccumulateBy sums AllocationSets based on the resolution given. The resolution given is subject to the scale used for the AllocationSets.
-// Resolutions not evenly divisible by the AllocationSetRange window durations Accumulate sets until a sum greater than or equal to the resolution is met,
-// at which point AccumulateBy will start summing from 0 until the requested resolution is met again.
-// If the requested resolution is smaller than the window of an AllocationSet then the resolution will default to the duration of a set.
-// Resolutions larger than the duration of the entire AllocationSetRange will default to the duration of the range.
-func (asr *AllocationSetRange) AccumulateBy(resolution time.Duration) (*AllocationSetRange, error) {
-	allocSetRange := NewAllocationSetRange()
-	var allocSet *AllocationSet
+// Accumulate sums AllocationSets based on the AccumulateOption (calendar week or calendar month).
+// The accumulated set is determined by the start of the window of the allocation set.
+func (asr *AllocationSetRange) Accumulate(accumulateBy AccumulateOption, clone bool) (*AllocationSetRange, error) {
```

Comment:
Is this `clone` flag the only option here? I'm afraid that we don't want to publicly expose the option to users. Would it be worth a quick conversation tomorrow morning to go over this?

Reply:
Lets do that. I was hoping to not force the user the penalties of cloning - but if we are fine with that, we can just make it always a clone. With the main penalty being the time to copy data (memory space should be minimal).